include/superior_mysqlpp/converters.hpp

```diff
@@ -14,22 +14,45 @@
 
 namespace SuperiorMySqlpp { namespace Converters
 {
-    template<typename T>
-    inline std::enable_if_t<std::is_integral<T>::value, T> to(const char* str, unsigned int length)
-    {
-        return toInteger<T>(str, length);
+    namespace detail {
+        template<typename First, std::size_t>
```

Comment:
It won't work in this way, you need to have that template specialization lookup implemented like so, to ensure proper sfinae.
```c++
        // not needed in c++17
        template<typename... Ts> struct make_void { typedef void type;};
        template<typename... Ts> using void_t = typename make_void<Ts...>::type;

        template<typename T>
        using get_to_method_type = decltype(&To<T>::operator());
        
        template<typename T, typename = void>
        struct has_to_overload : std::false_type {};
        
        template<typename T>
        struct has_to_overload<T, void_t<get_to_method_type<T>>> : std::true_type {};
```